rust/lance/src/index/cache.rs

```diff
@@ -22,25 +23,33 @@ use super::vector::VectorIndex;
 use std::sync::atomic::{AtomicU64, Ordering};
 
 #[derive(Debug, Default)]
-pub struct CacheStats {
+struct CacheStats {
     hits: AtomicU64,
     misses: AtomicU64,
 }
 
 impl CacheStats {
-    pub fn record_hit(&self) {
+    fn record_hit(&self) {
         self.hits.fetch_add(1, Ordering::Relaxed);
     }
 
-    pub fn record_miss(&self) {
+    fn record_miss(&self) {
         self.misses.fetch_add(1, Ordering::Relaxed);
     }
 }
 
 #[derive(Clone)]
 pub struct IndexCache {
+    // TODO: Can we merge these two caches into one for uniform memory management?
```

Comment:
Downcasting/upcasting is weird.  You can try.  I got spun around and gave up on it at the time simply because the PR had lots of other stuff.